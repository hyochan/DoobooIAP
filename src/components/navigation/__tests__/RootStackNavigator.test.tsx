import 'react-native';

import * as firebase from 'firebase';

import React, { ReactElement } from 'react';
import {
  act,
  render,
  wait,
} from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import StackNavigator from '../RootStackNavigator';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let props: any;
let component: ReactElement;

const onAuthStateChanged = jest.fn();

const getRedirectResult = jest.fn(() => {
  return Promise.resolve({
    user: {
      displayName: 'redirectResultTestDisplayName',
      email: 'redirectTest@test.com',
      emailVerified: true,
    },
  });
});

const sendEmailVerification = jest.fn(() => {
  return Promise.resolve('result of sendEmailVerification');
});

const sendPasswordResetEmail = jest.fn(() => Promise.resolve());

const createUserWithEmailAndPassword = jest.fn(() => {
  return Promise.resolve('result of createUserWithEmailAndPassword');
});

const signInWithEmailAndPassword = jest.fn(() => {
  return Promise.resolve('result of signInWithEmailAndPassword');
});

const signInWithRedirect = jest.fn(() => {
  return Promise.resolve('result of signInWithRedirect');
});

const initializeApp = jest
  .spyOn(firebase, 'initializeApp')
  .mockImplementation((): any => {
    return {
      auth: (): object => {
        return {
          createUserWithEmailAndPassword,
          signInWithEmailAndPassword,
          currentUser: {
            sendEmailVerification,
          },
          signInWithRedirect,
        };
      },
    };
  });

jest.spyOn(firebase, 'auth').mockImplementation((): any => {
  return {
    onAuthStateChanged,
    currentUser: {
      displayName: 'testDisplayName',
      email: 'test@test.com',
      emailVerified: true,
    },
    getRedirectResult,
    sendPasswordResetEmail,
  };
});

describe('[Stack] navigator', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(
      <StackNavigator {...props} />,
    );
  });

  it('should renders without crashing', async () => {
    const { container } = render(component);
    await act(() => wait());
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
