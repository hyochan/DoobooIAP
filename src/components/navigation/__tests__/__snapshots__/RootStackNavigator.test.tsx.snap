// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Stack] navigator should renders without crashing 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={true}
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 0.01,
                    "restSpeedThreshold": 0.01,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 0.01,
                    "restSpeedThreshold": 0.01,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <SafeAreaView
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#ffffff",
                              "flexBasis": 0,
                              "flexDirection": "column",
                              "flexGrow": 1,
                              "flexShrink": 1,
                              "justifyContent": "center",
                            },
                          ]
                        }
                      >
                        <ScrollView
                          contentContainerStyle={
                            Object {
                              "alignItems": "center",
                            }
                          }
                          style={
                            Object {
                              "width": "100%",
                            }
                          }
                        >
                          <TouchableOpacity
                            activeOpacity={0.2}
                            style={
                              Object {
                                "marginTop": 120,
                              }
                            }
                          >
                            <Image
                              source={
                                Object {
                                  "canInstrument": true,
                                  "createTransformer": [Function],
                                  "getCacheKey": [Function],
                                  "process": [Function],
                                }
                              }
                              style={
                                Object {
                                  "height": 80,
                                  "width": 300,
                                }
                              }
                            />
                          </TouchableOpacity>
                          <View
                            style={
                              Object {
                                "marginHorizontal": 48,
                              }
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "stretch",
                                    "flexDirection": "column",
                                  },
                                  Object {
                                    "marginTop": 60,
                                  },
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#b9b9c4",
                                      "fontSize": 14,
                                      "fontWeight": "500",
                                      "marginBottom": 5,
                                    },
                                    Object {
                                      "fontSize": 14,
                                    },
                                    null,
                                  ]
                                }
                              >
                                Email
                              </Text>
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                    },
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "justifyContent": "space-between",
                                        "width": "100%",
                                      },
                                      Object {
                                        "borderBottomWidth": 0.6,
                                        "borderColor": "#151A25",
                                      },
                                      undefined,
                                      null,
                                    ]
                                  }
                                >
                                  <TextInput
                                    allowFontScaling={true}
                                    autoCapitalize="none"
                                    placeholder="Please write email address"
                                    placeholderTextColor="#676c7a"
                                    rejectResponderTermination={true}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#2c374e",
                                          "flexBasis": 0,
                                          "flexGrow": 1,
                                          "flexShrink": 1,
                                          "fontSize": 15,
                                          "fontWeight": "500",
                                          "height": "100%",
                                          "paddingBottom": 15,
                                          "paddingTop": 15,
                                        },
                                        Object {
                                          "color": "#151A25",
                                          "fontSize": 16,
                                        },
                                      ]
                                    }
                                    testID="input-email"
                                    underlineColorAndroid="transparent"
                                    value=""
                                  />
                                </View>
                              </View>
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "stretch",
                                    "flexDirection": "column",
                                  },
                                  Object {
                                    "marginTop": 32,
                                  },
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#b9b9c4",
                                      "fontSize": 14,
                                      "fontWeight": "500",
                                      "marginBottom": 5,
                                    },
                                    Object {
                                      "fontSize": 14,
                                    },
                                    null,
                                  ]
                                }
                              >
                                Password
                              </Text>
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                    },
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "justifyContent": "space-between",
                                        "width": "100%",
                                      },
                                      Object {
                                        "borderBottomWidth": 0.6,
                                        "borderColor": "#151A25",
                                      },
                                      undefined,
                                      null,
                                    ]
                                  }
                                >
                                  <TextInput
                                    allowFontScaling={true}
                                    autoCapitalize="none"
                                    placeholder="******"
                                    placeholderTextColor="#676c7a"
                                    rejectResponderTermination={true}
                                    secureTextEntry={true}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#2c374e",
                                          "flexBasis": 0,
                                          "flexGrow": 1,
                                          "flexShrink": 1,
                                          "fontSize": 15,
                                          "fontWeight": "500",
                                          "height": "100%",
                                          "paddingBottom": 15,
                                          "paddingTop": 15,
                                        },
                                        Object {
                                          "color": "#151A25",
                                          "fontSize": 16,
                                        },
                                      ]
                                    }
                                    testID="input-password"
                                    underlineColorAndroid="transparent"
                                    value=""
                                  />
                                </View>
                              </View>
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "flexDirection": "row",
                                    "marginTop": 20,
                                    "width": "100%",
                                  },
                                ]
                              }
                            >
                              <TouchableOpacity
                                activeOpacity={0.2}
                                style={
                                  Object {
                                    "flex": 1,
                                    "flexDirection": "row",
                                    "height": 52,
                                    "justifyContent": "center",
                                  }
                                }
                                testID="btn-sign-up"
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "alignSelf": "center",
                                        "borderColor": "blue",
                                        "flexDirection": "row",
                                        "height": 52,
                                        "justifyContent": "center",
                                      },
                                      Object {
                                        "backgroundColor": "#ffffff",
                                        "borderColor": "#A750B9",
                                        "borderWidth": 1,
                                        "width": "100%",
                                      },
                                    ]
                                  }
                                >
                                  <Text
                                    style={
                                      Array [
                                        Object {
                                          "color": "#069ccd",
                                          "fontSize": 16,
                                        },
                                        Object {
                                          "color": "#A750B9",
                                          "fontSize": 14,
                                          "fontWeight": "bold",
                                        },
                                      ]
                                    }
                                  >
                                    Sign Up
                                  </Text>
                                </View>
                              </TouchableOpacity>
                              <View
                                style={
                                  Object {
                                    "width": 20,
                                  }
                                }
                              />
                              <TouchableOpacity
                                activeOpacity={0.2}
                                style={
                                  Object {
                                    "backgroundColor": "#A750B9",
                                    "flex": 1,
                                    "flexDirection": "row",
                                    "height": 52,
                                    "justifyContent": "center",
                                  }
                                }
                                testID="btn-sign-in"
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "alignSelf": "center",
                                        "borderColor": "blue",
                                        "flexDirection": "row",
                                        "height": 52,
                                        "justifyContent": "center",
                                      },
                                    ]
                                  }
                                >
                                  <Text
                                    style={
                                      Array [
                                        Object {
                                          "color": "#069ccd",
                                          "fontSize": 16,
                                        },
                                        Object {
                                          "color": "#ffffff",
                                          "fontSize": 14,
                                          "fontWeight": "bold",
                                        },
                                      ]
                                    }
                                  >
                                    Login
                                  </Text>
                                </View>
                              </TouchableOpacity>
                            </View>
                            <TouchableOpacity
                              activeOpacity={0.2}
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "center",
                                    "marginBottom": 44,
                                    "paddingBottom": 20,
                                    "paddingLeft": 20,
                                    "paddingRight": 20,
                                    "paddingTop": 20,
                                  },
                                ]
                              }
                              testID="btn-find-pw"
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#1E6EFA",
                                      "textDecorationLine": "underline",
                                    },
                                  ]
                                }
                              >
                                Forgot password?
                              </Text>
                            </TouchableOpacity>
                            <View
                              style={
                                Array [
                                  Object {
                                    "flexDirection": "row",
                                    "flexWrap": "wrap",
                                    "justifyContent": "center",
                                    "paddingBottom": 40,
                                    "paddingLeft": 0,
                                    "paddingRight": 0,
                                    "paddingTop": 0,
                                  },
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#777",
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                We consider that you agree with 
                              </Text>
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#1E6EFA",
                                      "lineHeight": 22,
                                      "textDecorationLine": "underline",
                                    },
                                  ]
                                }
                                testID="btn-terms"
                              >
                                Terms for Agreement
                              </Text>
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#777",
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                 and 
                              </Text>
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#1E6EFA",
                                      "lineHeight": 22,
                                      "textDecorationLine": "underline",
                                    },
                                  ]
                                }
                                testID="btn-privacy"
                              >
                                Privacy Policy
                              </Text>
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#777",
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                 by going onto next step.
                              </Text>
                            </View>
                          </View>
                        </ScrollView>
                      </SafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          pointerEvents="box-none"
          style={null}
        />
      </View>
    </RNCSafeAreaView>
  </View>
</View>
`;
